tests/core/test_okta.py

```diff
@@ -211,7 +213,101 @@ def test_okta_mfa_challenge(
             responses.POST,
             'https://organization.okta.com/api/v1/authn',
-            json=json.loads(AUTH_MFA_RESPONSE)
+            json=json.loads(AUTH_MFA_PUSH_RESPONSE)
+        )
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/authn/factors/id/verify',
+            json=json.loads(MFA_WAITING_RESPONSE)
+        )
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/authn/factors/id/lifecycle/activate/poll',
+            json=json.loads(AUTH_TOKEN_RESPONSE)
+        )
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/sessions',
+            json=json.loads(SESSION_RESPONSE)
+        )
+
+        okta = Okta(
+            user_name="user_name",
+            user_pass="user_pass",
+            organization="organization.okta.com"
+        )
+
+        self.assertEqual(okta.okta_single_use_token, "single_use_token")
+        self.assertEqual(okta.organization, "organization.okta.com")
+        self.assertEqual(okta.okta_session_id, "session_token")
+
+    @patch('aws_okta_processor.core.okta.input')
+    @patch('aws_okta_processor.core.okta.os.chmod')
+    @patch('aws_okta_processor.core.okta.open')
+    @patch('aws_okta_processor.core.okta.os.makedirs')
+    @patch('aws_okta_processor.core.okta.print_tty')
+    @responses.activate
+    def test_okta_mfa_totp_challenge(
+            self,
+            mock_print_tty,
+            mock_makedirs,
+            mock_open,
+            mock_chmod,
+            mock_input
+    ):
+        mock_input.return_value = "123456"
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/authn',
+            json=json.loads(AUTH_MFA_TOTP_RESPONSE)
+        )
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/authn/factors/id/verify',
+            json=json.loads(AUTH_TOKEN_RESPONSE)
+        )
+
+        responses.add(
+            responses.POST,
+            'https://organization.okta.com/api/v1/sessions',
+            json=json.loads(SESSION_RESPONSE)
+        )
+
+        okta = Okta(
+            user_name="user_name",
+            user_pass="user_pass",
+            organization="organization.okta.com"
+        )
+
+        self.assertEqual(okta.okta_single_use_token, "single_use_token")
+        self.assertEqual(okta.organization, "organization.okta.com")
+        self.assertEqual(okta.okta_session_id, "session_token")
+
+    @patch('aws_okta_processor.core.prompt.input')
+    @patch('aws_okta_processor.core.okta.os.chmod')
+    @patch('aws_okta_processor.core.okta.open')
+    @patch('aws_okta_processor.core.okta.os.makedirs')
+    @patch('aws_okta_processor.core.okta.print_tty')
+    @responses.activate
+    def test_okta_mfa_push_multiple_factor_challenge(
+            self,
+            mock_print_tty,
```

Comment:
I noticed that there is a print leaking through here from this test; I was able to eliminate it (and cut down on the number of mocks passed to the test!) with this change:
```diff
@@ -292,11 +292,11 @@ class TestOkta(TestBase):
     @patch('aws_okta_processor.core.okta.os.chmod')
     @patch('aws_okta_processor.core.okta.open')
     @patch('aws_okta_processor.core.okta.os.makedirs')
-    @patch('aws_okta_processor.core.okta.print_tty')
+    @patch('aws_okta_processor.core.okta.print_tty', new=MagicMock())
+    @patch('aws_okta_processor.core.prompt.print_tty', new=MagicMock())
     @responses.activate
     def test_okta_mfa_push_multiple_factor_challenge(
             self,
-            mock_print_tty,
             mock_makedirs,
             mock_open,
             mock_chmod,
```
Aside from that, I think this is looking great. Test coverage looks good enough to me. Nice work!

Reply:
Appreciate that! Had no idea how to suppress that leakage... Done now!